static/app/components/hovercard.tsx

```diff
@@ -149,91 +119,160 @@ class Hovercard extends React.Component<Props, State> {
         enabled: true,
         boundariesElement: 'viewport',
       },
-      ...(modifiers || {}),
+      ...(props.modifiers || {}),
     };
+    return modifiers;
+  }, [props.modifiers]);
 
-    const visible = show !== undefined ? show : this.state.visible;
-    const hoverProps =
-      show !== undefined
-        ? {}
-        : {onMouseEnter: this.handleToggleOn, onMouseLeave: this.handleToggleOff};
-
-    return (
-      <Manager>
-        <Reference>
-          {({ref}) => (
-            <span
-              ref={ref}
-              aria-describedby={this.tooltipId}
-              className={containerClassName}
-              {...hoverProps}
-            >
-              {this.props.children}
-            </span>
-          )}
-        </Reference>
-        {visible &&
-          (header || body) &&
-          ReactDOM.createPortal(
-            <Popper placement={position} modifiers={popperModifiers}>
-              {({ref, style, placement, arrowProps, scheduleUpdate}) => {
-                this.scheduleUpdate = scheduleUpdate;
-                return (
-                  <StyledHovercard
-                    id={this.tooltipId}
-                    visible={visible}
-                    ref={ref}
-                    style={style}
-                    placement={placement as Direction}
-                    offset={offset}
-                    className={cx}
-                    {...hoverProps}
-                  >
-                    {header && <Header>{header}</Header>}
-                    {body && <Body className={bodyClassName}>{body}</Body>}
-                    <HovercardArrow
-                      ref={arrowProps.ref}
-                      style={arrowProps.style}
-                      placement={placement as Direction}
-                      tipColor={tipColor}
-                      tipBorderColor={tipBorderColor}
-                    />
-                  </StyledHovercard>
-                );
-              }}
-            </Popper>,
-            this.portalEl
-          )}
-      </Manager>
-    );
-  }
+  const isVisible = React.useMemo(() => {
+    // If show is not set, then visibility state is uncontrolled
+    if (props.show === undefined) {
+      return visible;
+    }
+    return props.show;
+  }, [props.show, visible]);
+
+  const hoverProps = React.useMemo((): Pick<
```

Comment:
do we need `useMemo` here?

Reply:
Same thing here, as the comment above, we are creating a new object reference here otherwise